src/Keg.sol

```diff
@@ -64,126 +239,108 @@ contract Keg is LibNote {
-
-    // Two-way mapping tracks delegates
-    mapping (address => address) public pals;   // Delegate -> Original
-    mapping (address => address) public buds;   // Original -> Delegate
+    // Define payout ratios
+    mapping (bytes32 => Pint[]) public flights;       // The Pint definitions
 
     // --- Events ---
     event NewBrewMaster(address brewmaster);
     event RetiredBrewMaster(address brewmaster);
-    event BrewBeer(uint256 beer);
     event PourBeer(address bartender, uint256 beer);
-    event HoldMyBeerBro(address indexed owner, address bud);
-    event ByeFelicia(address indexed owner, address bud);
-    event JustASip(address bum, uint256 beer);
-    event DownTheHatch(address bum, uint256 beer);
+    event OrdersUp(bytes32 flight);
+    event OrderRevoked(bytes32 flight);
 
-    constructor(address vat_, address vow_) public {
+    constructor(address vat_) public {
         wards[msg.sender] = 1;
         vat = VatLike(vat_);
-        vow = vow_;
-        beer = 0;
-    }
-
-    // Suck from the vat to the keg to allow for a pool of funds
-    function brew(uint wad) external note auth stoppable {
-    	vat.suck(address(vow), address(this), mul(wad, RAY));
-        emit BrewBeer(wad);
     }
 
     // Credits people with rights to withdraw funds from the pool
-    function pour(address[] calldata bums, uint[] calldata wad) external note auth stoppable {
-        require(bums.length == wad.length, "Keg/unequal-payees-and-amounts");
+    function pour(address[] calldata bums, uint256[] calldata rad) external note stoppable {
+        require(bums.length == rad.length, "Keg/unequal-payees-and-amounts");
         require(bums.length > 0, "Keg/no-bums");
-        uint suds = 0;
-        for (uint i = 0; i < wad.length; i++) {
+        for (uint256 i = 0; i < rad.length; i++) {
             require(bums[i] != address(0), "Keg/no-address-0");
-            mugs[bums[i]] = add(mugs[bums[i]], wad[i]);
-            suds          = add(suds, wad[i]);
-            emit PourBeer(bums[i], wad[i]);
+            vat.move(msg.sender, bums[i], rad[i]);
+            emit PourBeer(bums[i], rad[i]);
         }
-        beer = add(beer, suds);
-        require(vat.dai(address(this)) == mul(beer, RAY), "Keg/pour-not-equal-to-brew");
     }
 
-    // User delegates compensation to another address
-    function pass(address bud) external {
-        require(bud != msg.sender, "Keg/cannot_delegate_to_self");
-        require(pals[bud] == address(0), "Keg/bud-already-has-a-pal");
-        // Remove existing delegate
-        if (buds[msg.sender] != address(0)) yank();
-        // Original addr -> delegated addr
-        buds[msg.sender] = bud;
-        // Delegated addr -> original addr
-        pals[bud] = msg.sender;
-        emit HoldMyBeerBro(msg.sender, bud);
-    }
+    // Credits people with rights to withdraw funds from the pool using a preset flight
+    function pour(bytes32 flight, uint256 rad) external note stoppable {
```

Comment:
We'll discuss this more after this is merged, but it might be worth making a pour function that will automatically distribute funds across **all** flights. I don't think the FlapTap should have a global variable for `flight`.

Reply:
Sure happy to discuss.